tests/cli/main.cpp

```diff
@@ -197,102 +199,118 @@ class client_connection
+
+         // import Nathan account
+         BOOST_TEST_MESSAGE("Importing nathan key");
+         BOOST_CHECK_EQUAL(nathan_keys[0], "5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3");
+         BOOST_CHECK(con.wallet_api_ptr->import_key("nathan", nathan_keys[0]));
+      } catch( fc::exception& e ) {
+         edump((e.to_detail_string()));
+         throw;
+      }
+   }
 
-   } catch( fc::exception& e ) {
-      edump((e.to_detail_string()));
-      throw;
+   ~cli_fixture()
+   {
+      BOOST_TEST_MESSAGE("Cleanup cli_wallet::boost_fixture_test_case");
+
+      // wait for everything to finish up
+      fc::usleep(fc::seconds(1));
+        
+      app1->shutdown();
    }
-   app1->shutdown();
+};
+
+///////////////////////////////
+// Tests
+///////////////////////////////
+
+////////////////
+// Start a server and connect using the same calls as the CLI
+////////////////
+BOOST_FIXTURE_TEST_CASE( cli_connect, cli_fixture )
+{
+   BOOST_TEST_MESSAGE("Testing wallet connection.");
 }
 
 ////////////////
 // Start a server and connect using the same calls as the CLI
 // Quit wallet and be sure that file was saved correctly
 ////////////////
-BOOST_AUTO_TEST_CASE( cli_quit )
+BOOST_FIXTURE_TEST_CASE( cli_quit, cli_fixture )
 {
-   using namespace graphene::chain;
-   using namespace graphene::app;
-   std::shared_ptr<graphene::app::application> app1;
-   try {
-      BOOST_TEST_MESSAGE("Testing wallet connection and quit command.");
+   BOOST_TEST_MESSAGE("Testing wallet connection and quit command.");
+   BOOST_CHECK_THROW( con.wallet_api_ptr->quit(), fc::canceled_exception );
+}
 
-      fc::temp_directory app_dir ( graphene::utilities::temp_directory_path() );
+BOOST_FIXTURE_TEST_CASE( upgrade_nathan_account, cli_fixture )
+{
+   try
+   {
+      BOOST_TEST_MESSAGE("Upgrade Nathan's account");
 
-      int server_port_number = 0;
-      app1 = start_application(app_dir, server_port_number);
+      account_object nathan_acct_before_upgrade, nathan_acct_after_upgrade;
+      std::vector<signed_transaction> import_txs;
+      signed_transaction upgrade_tx;
 
-      // connect to the server
-      client_connection con(app1, app_dir, server_port_number);
+      BOOST_TEST_MESSAGE("Importing nathan's balance");
+      import_txs = con.wallet_api_ptr->import_balance("nathan", nathan_keys, true);
+      nathan_acct_before_upgrade = con.wallet_api_ptr->get_account("nathan");
 
-      BOOST_CHECK_THROW( con.wallet_api_ptr->quit(), fc::canceled_exception );
+      // upgrade nathan
+      BOOST_TEST_MESSAGE("Upgrading Nathan to LTM");
+      upgrade_tx = con.wallet_api_ptr->upgrade_account("nathan", true);
+      nathan_acct_after_upgrade = con.wallet_api_ptr->get_account("nathan");
+
+      // verify that the upgrade was successful
+      BOOST_CHECK_PREDICATE( std::not_equal_to<uint32_t>(), (nathan_acct_before_upgrade.membership_expiration_date.sec_since_epoch())(nathan_acct_after_upgrade.membership_expiration_date.sec_since_epoch()) );
```

Comment:
This line is too long to fit in my screen as well.

By the way there were some long lines that didn't change, so I'm unable to comment on them. Please help wrap them as well.

Reply:
ok sure ! I will format the whole file right now as required !